### ISSUE TITLE
input, child element not rendered to string

### ISSUE THREAD

**Issue by jogibear9988** [AngleSharp/AngleSharp#727]
I create a input element and add a child element to it (via "appendChild").

when I now create the html doc via 

    htmlDocument.Body.ToHtml(new PrettyMarkupFormatter())

the child of the input is not rendered to html

forgot, the input is of type text.

If I add a child element to it in chrome console it works.

**Comment by jogibear9988** [comment]
Okay, seems a input should not have children. I chrome it only works if I use appendChild.

If I do this:
 
     $0.innerHTML="<input><div></div></input>"

div is also added, but behind the input

**Comment by jogibear9988** [comment]
That means, should it be ignored, or added behind?

**Comment by FlorianRappl** [comment]
Hm sorry did not receive any notification for this one @jogibear9988.

Yes, by the spec `input` does not have children and is self-closing (i.e., closing tag is ignored and any "children" are siblings). However, when you self-manipulate the DOM it should still "work" (like Chrome does). Note, however, that it only works in the serialization, for rendering / any functional consequence it does not have any impact.

The reason for not showing the "attached" children for input (or any self-closing element) is that AngleSharp respects the flags of the elements when serializing. I tested this in Chrome and the serialization is behaving the same as in AngleSharp, so I wonder if you did not look at the childNodes in Chrome (which should deliver 1, like in AngleSharp - the `div`).

So tl;dr: When parsing it should be a sibling, when `appendChild` you should not see anything in a serialization, but via, e.g., `childNodes`.

**Comment by jogibear9988** [comment]
As I said, Chrome behaves the same, so I close it.
And as you say, it is self closing, so that explains everything!
